python/ray/dag/compiled_dag_node.py

```diff
@@ -958,23 +994,34 @@ def _preprocess(self) -> None:
         # custom NCCL groups for a set of actors, only one is cached.
         actors_to_nccl_group_id: Dict[FrozenSet["ray.actor.ActorHandle"], str] = {}
 
-        # If a custom NCCL group is specified for P2P actors, initialize and cache
-        # the NCCL group ID.
-        if nccl_actors_p2p and self._custom_nccl_group_p2p:
-            if not set(nccl_actors_p2p).issubset(
-                set(self._custom_nccl_group_p2p.get_actor_handles())
-            ):
+        # If a custom NCCL group is specified for P2P actors, initialize and
+        # cache the NCCL group ID.
+        for (
+            custom_nccl_group,
+            nodes,
```

Comment:
nit: let's call this `dag_nodes` or `senders_and_receivers`

Reply:
Done.